fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java

```diff
@@ -0,0 +1,249 @@
+    // members below updated when job state changed to etl
+    private long etlStartTimestamp = -1;
+    // for spark yarn
+    private String appId = "";
+    // spark job outputPath
+    private String etlOutputPath = "";
+    // members below updated when job state changed to loading
+    // { tableId.partitionId.indexId.bucket.schemaHash -> (etlFilePath, etlFileSize) }
+    private Map<String, Pair<String, Long>> tabletMetaToFileInfo = Maps.newHashMap();
+
+    // --- members below not persist ---
+    // temporary use
+    // one SparkLoadJob has only one table to load
+    // hivedb.table for global dict
+    private String hiveTableName = "";
+    private ResourceDesc resourceDesc;
+    // for spark standalone
+    private SparkAppHandle sparkAppHandle;
+    // for straggler wait long time to commit transaction
+    private long quorumFinishTimestamp = -1;
+    // below for push task
+    private Map<Long, Set<Long>> tableToLoadPartitions = Maps.newHashMap();
+    //private Map<Long, PushBrokerScannerParams> indexToPushBrokerReaderParams = Maps.newHashMap();
+    private Map<Long, Integer> indexToSchemaHash = Maps.newHashMap();
+    private Map<Long, Map<Long, PushTask>> tabletToSentReplicaPushTask = Maps.newHashMap();
+    private Set<Long> finishedReplicas = Sets.newHashSet();
+    private Set<Long> quorumTablets = Sets.newHashSet();
+    private Set<Long> fullTablets = Sets.newHashSet();
+
+    // only for log replay
+    public SparkLoadJob() {
+        super();
+        jobType = EtlJobType.SPARK;
+    }
+
+    public SparkLoadJob(long dbId, String label, ResourceDesc resourceDesc, OriginStatement originStmt)
+            throws MetaNotFoundException {
+        super(dbId, label, originStmt);
+        this.resourceDesc = resourceDesc;
+        timeoutSecond = Config.spark_load_default_timeout_second;
+        jobType = EtlJobType.SPARK;
+    }
+
+    public String getHiveTableName() {
+        return hiveTableName;
+    }
+
+    @Override
+    protected void setJobProperties(Map<String, String> properties) throws DdlException {
+        super.setJobProperties(properties);
+
+        // set spark resource and broker desc
+        setResourceInfo();
+
+        // global dict
+        if (properties != null) {
+            if (properties.containsKey(BITMAP_DATA_PROPERTY)) {
+                hiveTableName = properties.get(BITMAP_DATA_PROPERTY);
+            }
+        }
+    }
+
+    /**
+     * merge system conf with load stmt
+     * @throws DdlException
+     */
+    private void setResourceInfo() throws DdlException {
+        // spark resource
+        String resourceName = resourceDesc.getName();
+        Resource oriResource = Catalog.getCurrentCatalog().getResourceMgr().getResource(resourceName);
+        if (oriResource == null) {
+            throw new DdlException("Resource does not exist. name: " + resourceName);
+        }
+        Preconditions.checkState(oriResource instanceof SparkResource);
```

Comment:
Why using `Preconditions` here?
It may be some other kind of resource.

Reply:
remove this check